sys/include/net/nanocoap.h

```diff
@@ -192,6 +193,9 @@ typedef struct {
 #ifdef MODULE_GCOAP
     uint32_t observe_value;                           /**< observe value           */
 #endif
+#ifdef MODULE_NANOCOAP_PKT_IPV6_ADDRESS
+    ipv6_addr_t source_address;                       /**< request source address  */
+#endif
```

Comment:
Maybe a different datastructure could be used for storing the ip address. (I just noticed that this field should be renamed to "target address".)